### PR TITLE
Order imbalance table model

### PR DESCRIPTION
**Pull request by spireken**
I originally proposed an OrderImbalanceIndicatorModelSubscriber class that would manage the subscription to a model, but I removed it because it didn't provide enough value in practice to make maintaining it worthwhile.

### CODE REVIEW

File: Applications/Spire/Include/Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        std::shared_ptr<OrderImbalanceIndicatorModel> source);
+
+      int get_row_size() const override;
+
+      int get_column_size() const override;
+
+      const std::any& at(int row, int column) const override;
+
+      boost::signals2::connection connect_operation_signal(
+        const typename OperationSignal::slot_type& slot) const override;
```
Comment (shaobinspire) [inline Applications/Spire/Include/Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.hpp:32]: Remove `typename`.

File: Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  return m_table.get_column_size();
+}
+
+const std::any& OrderImbalanceIndicatorTableModel::at(
+    int row, int column) const {
+  return m_table.at(row, column);
+}
+
+connection OrderImbalanceIndicatorTableModel::connect_operation_signal(
+    const typename OperationSignal::slot_type& slot) const {
```
Comment (shaobinspire) [inline Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp:32]: Remove `typename`.

File: Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.hpp"
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+OrderImbalanceIndicatorTableModel::OrderImbalanceIndicatorTableModel(
+    TimeInterval interval,
+    std::shared_ptr<OrderImbalanceIndicatorModel> source)
+    : m_source(std::move(source)) {
```
Comment (shaobinspire) [inline Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp:10]: Why is the source stored in the class? It isn't used in other member functions except in the constructor.

**Review by shaobinspire — CHANGES_REQUESTED** [review]

File: Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp (hunk 3 above)
Comment (spireken) [reply Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorTableModel.cpp:10]: It keeps a pointer to it so the subscription won't be cancelled if whatever else is holding a pointer to the OrderImbalanceIndicatorModel is destroyed.

**Review by shaobinspire — APPROVED** [review]

File: Applications/Spire/Include/Spire/SpireTester/SpireTester.hpp [hunk 4]
```diff
@@ -29,6 +29,12 @@ namespace Spire {
       });
     app.exec();
   }
+
+  /**
+   * Processes events in the Qt event loop until the given predicate returns
+   * true.
+   */
+  void wait_until(const std::function<bool ()>& predicate);
```
Comment (kranar) [inline Applications/Spire/Include/Spire/SpireTester/SpireTester.hpp:37]: Changes to the public API need to be approved. Remove this function.

Comment (spireken) [reply Applications/Spire/Include/Spire/SpireTester/SpireTester.hpp:37]: I'm looking into a TestOrderImbalanceIndicatorModel than can resolve subscriptions manually.

**Review by kranar — APPROVED** [review]

**Pull request merged**